pipelines/build/common/openjdk_build_pipeline.groovy

```diff
@@ -58,6 +59,22 @@ class Build {
     String buildSource = ""
     Map variantVersion = [:]
 
+    // Declare timeouts for each critical stage (unit is HOURS)
+    Map buildTimeouts = [
+        API_REQUEST_TIMEOUT : 1,
+        NODE_CLEAN_TIMEOUT : 1,
+        NODE_CHECKOUT_TIMEOUT : 1,
+        BUILD_JDK_TIMEOUT : 6,
+        BUILD_ARCHIVE_TIMEOUT : 3,
+        AIX_CLEAN_TIMEOUT : 1,
+        MASTER_CLEAN_TIMEOUT : 1,
+        DOCKER_CHECKOUT_TIMEOUT : 1,
+        DOCKER_PULL_TIMEOUT : 2,
+        SIGN_JOB_TIMEOUT : 2,
+        TEST_JOBS_TIMEOUT : 10,
```

Comment:
Since I don't see any test team members on this PR yet I suggest we run this by @smlambert and @llxia. I know the test pipelines have their own, configurable timeout value. If we didn't have one for test would it cause a timeout "blindspot" or would it get covered in a higher level timeout?

Reply:
Thanks @AdamBrousseau - we do in fact manage our own timeouts in the test pipelines, because philosophically we believe each stage and step in the pipeline should manage their own timeouts and as there are valid cases where the default of 10hrs is different.

If the build pipelines want to enforce a 10hr limit, this PR should pass TEST_JOBS_TIMEOUT as a parameter in [openjdk_build_pipeline.groovy L224-L230](https://github.com/AdoptOpenJDK/openjdk-build/blob/master/pipelines/build/common/openjdk_build_pipeline.groovy#L224-L230).  The parameter name to set is called [TIME_LIMIT](https://github.com/AdoptOpenJDK/openjdk-tests/blob/master/buildenv/jenkins/JenkinsfileBase#L660).  

Centrally managing and hardcoding all of these timeouts from a build pipeline makes this build pipeline code very closely tied to the performance of the Adopt build/test farm.  I would expect a _future request_, if we are building flexible/portable pipeline code will be to allow for all of these settings to be tune-able (without having to maintain a fork of this repo).

Reply:
If the test pipelines have their own timeouts, I’m leaning towards just removing the timeouts I want to introduce from openjdk_build_pipeline.groovy completely. I don’t really want to have a clash of timeouts that counter each other